.github/workflows/perf_analysis.yml

```diff
@@ -0,0 +1,83 @@
+name: perf_analysis
+
+on:
+  push:
+    branches:
+      - 'master'
+      - 'release/**'
+  schedule:
+    # Once a day at 02:00, should be later than `perf_micro.yml`.
+    - cron: '0 2 * * *'
+
+concurrency:
+  # Update of a developer branch cancels the previously scheduled workflow
+  # run for this branch. However, the 'master' branch, release branch, and
+  # tag workflow runs are never canceled.
+  #
+  # We use a trick here: define the concurrency group as 'workflow run ID' +
+  # 'workflow run attempt' because it is a unique combination for any run.
+  # So it effectively discards grouping.
+  #
+  # Important: we cannot use `github.sha` as a unique identifier because
+  # pushing a tag may cancel a run that works on a branch push event.
+  group: ${{ (
+    github.ref == 'refs/heads/master' ||
+    startsWith(github.ref, 'refs/heads/release/') ||
+    startsWith(github.ref, 'refs/tags/')) &&
+    format('{0}-{1}', github.run_id, github.run_attempt) ||
+    format('{0}-{1}-perf-analysis', github.workflow, github.ref) }}
+  cancel-in-progress: true
+
+jobs:
+  perf_analysis:
+    runs-on:
+      - self-hosted
+      - Linux
+      - x86_64
+
+    timeout-minutes: 60
+
+    env:
+      PERF_BRANCH: ${{ github.event.pull_request.head.ref || github.ref_name }}
+      PERF_COMMIT: ${{ github.event.pull_request.head.sha || github.sha }}
+      INFLUXDB_URL: ${{ secrets.INFLUXDB_URL }}
```

Comment:
I'm not sure that these variables should be in the environment -- in that case, they may be printed (or sended as a body to the remote server) from the changed script. It's better to use them as is inside the curl command, which downloads the data.

Reply:
But IINM they still can be leaked if someone updates the script, since it runs on every pull request. Maybe it is better to restrict this workflow by `pull_request_target` triggered by the `performance` label that can be set only by maintainers? What do you think?

Reply:
Honestly, I don't get you. The workflow is not intended to run in pull requests, it should be started on push to master and release branches and also nightly. The event `pull_request` has been added for debug purposes in a temporary commit that should be removed before merge.

Reply:
Thanks for the clarification. I missed the part that this comment is temporary.